src/Projections/ProjectionHandler.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSauce\EventSourcing\Projections;
+
+use EventSauce\EventSourcing\Message;
+use EventSauce\EventSourcing\MessageConsumer;
+use EventSauce\EventSourcing\Subscriptions\Checkpoint;
+use EventSauce\EventSourcing\Subscriptions\SubscriptionProvider;
+
+class ProjectionHandler
+{
+    public function __construct(
+        private SubscriptionProvider $subscription,
+        private MessageConsumer $consumer,
+        private ProjectionStatusRepository $projectionStatusRepository,
+    ) {
+    }
+
+    public function handle(ProjectionId $projectionId, Checkpoint $initialCheckpoint): void
```

Comment:
this method probably need a strategy to deal with consumer failures

Reply:
Hmm yeah, agree. Not sure what the best strategy would be 🤔 
Initial thought would be to wrap the `consumer->handle` in a try-catch, and on failure commit the checkpoint at that point in time (before the message was processed) to the repository store. 

Calculating the checkpoint would be hard/impossible though, due to the partitioning of subscribers. 

Any ideas on what strategy would work best here?